### PR TITLE
Add initial cmd-template

### PR DESCRIPTION
**Pull request by denis-tingajkin**
Signed-off-by: denis-tingajkin <denis.tingajkin@xored.com>

### CODE REVIEW

File: .github/workflows/update-cmd-repositories.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+jobs:
+  update-cmd-nsmgr:
+    name: Update cmd-nsmgr
+    runs-on: ubuntu-latest
+    if: github.repository == 'networkservicemesh/cmd-template'
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          repository: [networkservicemesh/cmd-nsmgr, cmd-forwarder-sriov, cmd-forwarder-vppagent, cmd-registry-proxy-dns, cmd-registry-memory, cmd-exclude-prefixes-k8s]
```
Comment (denis-tingajkin) [inline .github/workflows/update-cmd-repositories.yaml:16]: Note: for each of this cmds should be created PR on update this repo :)

File: .github/workflows/update-cmd-repositories.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+jobs:
+  update-cmd-nsmgr:
+    name: Update cmd-nsmgr
+    runs-on: ubuntu-latest
+    if: github.repository == 'networkservicemesh/cmd-template'
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          repository: [networkservicemesh/cmd-nsmgr, networkservicemesh/cmd-forwarder-sriov, networkservicemesh/cmd-forwarder-vppagent, networkservicemesh/cmd-registry-proxy-dns, networkservicemesh/cmd-registry-memory, networkservicemesh/cmd-exclude-prefixes-k8s]
```
Comment (denis-tingajkin) [inline .github/workflows/update-cmd-repositories.yaml:16]: Do we need to start with one repository to check how it will work? :)

Comment (edwarnicke) [reply .github/workflows/update-cmd-repositories.yaml:16]: Lets definitely start with one and go from there :)

Comment (edwarnicke) [reply .github/workflows/update-cmd-repositories.yaml:16]: So... how do we avoid here having .github/workflows/update-cmd-repositories.yaml pushed out to all the cmd-* repos?

File: .golangci.yml [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+      printf:
+        funcs:
+          - (github.com/sirupsen/logrus.FieldLogger).Infof
+          - (github.com/sirupsen/logrus.FieldLogger).Warnf
+          - (github.com/sirupsen/logrus.FieldLogger).Errorf
+          - (github.com/sirupsen/logrus.FieldLogger).Fatalf
+  golint:
+    min-confidence: 0.8
+  goimports:
+    local-prefixes: github.com/networkservicemesh/sdk
```
Comment (edwarnicke) [inline .golangci.yml:33]: So... I'm pretty sure this is the wrong local-prefix...

Comment (denis-tingajkin) [reply .golangci.yml:33]: Nice catch! Will fix.

File: .github/workflows/update-cmd-repositories.yaml (hunk 2 above)
Comment (denis-tingajkin) [reply .github/workflows/update-cmd-repositories.yaml:16]: > how do we avoid here having .github/workflows/update-cmd-repositories.yaml pushed out to all the cmd-* repos?

I did not found an easy solution for this, can we just still use check that this action will work only if the repository name is has 'template'?

Comment (denis-tingajkin) [reply .github/workflows/update-cmd-repositories.yaml]: > Lets definitely start with one and go from there :)

Done.

File: PATTERN.md [hunk 4]
```diff
@@ -0,0 +1,283 @@
+# Intro
+
+cmd-forwarder-vppagent sets out a pattern for building other cmds in networkservicemesh.  This pattern is expected to
```
Comment (haiodo) [inline PATTERN.md:3]:
```suggestion
App sets out a pattern for building other cmds in networkservicemesh.  This pattern is expected to
```

File: PATTERN.md [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+It allows you to create a simple struct for your configuration, and in can unmarshal that from env variables.
+envconfig's [supported field types](https://github.com/kelseyhightower/envconfig#supported-struct-field-types) include
+anything that implements [encoding.TextUnmarshaler](https://golang.org/pkg/encoding/#TextUnmarshaler) and 
+[encoding.BinaryUnmarshaler](https://golang.org/pkg/encoding/#BinaryUnmarshaler).  This allows native use of handy things
+like url.URL and net.IP.  envconfig also interprets arrays as meaning that the env variable is a comma delimited list
+of the type of the array.
+
+[Example Config struct](https://github.com/networkservicemesh/cmd-forwarder-vppagent/blob/master/main.go#L46):
+```go
+type Config struct {
```
Comment (haiodo) [inline PATTERN.md:75]: Individual applications could have this changes, so best way is to make it empty? Thoughts

File: PATTERN.md [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+		logrus.Fatalf("error getting x509 source: %+v", err)
+	}
+```
+
+### Setup other needed things
+
+If you require a [vpp-agent](https://github.com/ligato/vpp-agent),
+[Example](https://github.com/networkservicemesh/cmd-forwarder-vppagent/blob/9a981c41e568582c3916f8143a826edcd7d8162e/main.go#L84):
+```go
+vppagentCC, vppagentErrCh := vppagent.StartAndDialContext(ctx)
```
Comment (haiodo) [inline PATTERN.md:119]: This is not required.

File: PATTERN.md (hunk 4 above)
Comment (denis-tingajkin) [reply PATTERN.md]: Finally, PATTERN.md is excluded

**Pull request merged**